I could use some advice on the test. I'm using a path to a file on my machine, then getting error:
```
    @patch('os.stat', Mock(return_value=FAKE))
    def test_filesize_class(self):
        self.assertTrue(
            setup_component(self.hass, 'sensor', {'sensor': self.config}))

        state = self.hass.states.get('sensor.homeassistant_v2db')

>       assert state.state == 37.99
E       AttributeError: 'NoneType' object has no attribute 'state'
```

Also probably theres a more appropriate way than adding the class FakeObj() ?
Cheers

--- a/tests/components/sensor/test_filesize.py
+++ b/tests/components/sensor/test_filesize.py
@@ -4,14 +4,9 @@ from unittest.mock import Mock, patch
 
 from homeassistant.components.sensor.filesize import CONF_FILE_PATHS
 from homeassistant.setup import setup_component
-from tests.common import get_test_home_assistant
+from tests.common import get_test_home_assistant, mock_registry
 
 PATH = '/Users/robincole/.homeassistant/home-assistant_v2.db'
-
-VALID_CONFIG = {
-    'platform': 'filesize',
-    CONF_FILE_PATHS: [PATH]
-    }
 
 
 class FakeObj():
@@ -26,21 +21,30 @@ FAKE = FakeObj(37990000, 1518126363.5514238, 33188)
 
 
 class TestFileSensor(unittest.TestCase):
-    """Test the File sensor."""
+    """Test the filesize sensor."""
 
     def setup_method(self, method):
         """Set up things to be run when tests are started."""
         self.hass = get_test_home_assistant()
-        self.config = VALID_CONFIG
+        mock_registry(self.hass)
 
     def teardown_method(self, method):
         """Stop everything that was started."""
         self.hass.stop()
 
+    @patch('os.path.isfile', Mock(return_value=True))
+    @patch('os.access', Mock(return_value=True))
     @patch('os.stat', Mock(return_value=FAKE))
     def test_filesize_class(self):
+
+        config = {
+            'sensor': {
+                'platform': 'filesize',
+                CONF_FILE_PATHS: [PATH]}
+        }
+
         self.assertTrue(
-            setup_component(self.hass, 'sensor', {'sensor': self.config}))
+            setup_component(self.hass, 'sensor', config))
 
         state = self.hass.states.get('sensor.homeassistant_v2db')
 

--- a/tests/components/sensor/test_filesize.py
+++ b/tests/components/sensor/test_filesize.py
@@ -6,8 +6,6 @@ from homeassistant.components.sensor.filesize import CONF_FILE_PATHS
 from homeassistant.setup import setup_component
 from tests.common import get_test_home_assistant, mock_registry
 
-PATH = '/Users/robincole/.homeassistant/home-assistant_v2.db'
-
 
 class FakeObj():
     '''Fake object for testing.'''
@@ -17,11 +15,10 @@ class FakeObj():
         self.st_mode = st_mode
 
 
-FAKE = FakeObj(37990000, 1518126363.5514238, 33188)
-
-
 class TestFileSensor(unittest.TestCase):
     """Test the filesize sensor."""
+
+    FAKE = FakeObj(37990000, 1518126363.5514238, 33188)
 
     def setup_method(self, method):
         """Set up things to be run when tests are started."""
@@ -32,24 +29,24 @@ class TestFileSensor(unittest.TestCase):
         """Stop everything that was started."""
         self.hass.stop()
 
-    @patch('os.path.isfile', Mock(return_value=True))
-    @patch('os.access', Mock(return_value=True))
     @patch('os.stat', Mock(return_value=FAKE))
     def test_filesize_class(self):
-
+        """Test the filesize class."""
         config = {
             'sensor': {
                 'platform': 'filesize',
-                CONF_FILE_PATHS: [PATH]}
+                CONF_FILE_PATHS: ['mock.file1']}
         }
 
         self.assertTrue(
             setup_component(self.hass, 'sensor', config))
 
-        state = self.hass.states.get('sensor.homeassistant_v2db')
+        assert len(self.hass.states.entity_ids()) == 0
 
-        assert state.state == 37.99
-        assert state.attributes.get('bytes') == 37990000
-        assert state.attributes.get(
-            'last_updated') == '2018-02-08 21:46:03.551424'
-        assert state.attributes.get('path') == PATH
+        #state = self.hass.states.get('sensor.test_api_streamspy')
+
+    #    assert state.state == 37.99
+    #    assert state.attributes.get('bytes') == 37990000
+    #    assert state.attributes.get(
+    #        'last_updated') == '2018-02-08 21:46:03.551424'
+    #    assert state.attributes.get('path') == PATH

--- a/tests/components/sensor/test_filesize.py
+++ b/tests/components/sensor/test_filesize.py
@@ -29,8 +29,7 @@ class TestFileSensor(unittest.TestCase):
         """Stop everything that was started."""
         self.hass.stop()
 
-    @patch('os.stat', Mock(return_value=FAKE))
-    def test_filesize_class(self):
+    def test_invalid_path(self):
         """Test the filesize class."""
         config = {
             'sensor': {
@@ -42,6 +41,20 @@ class TestFileSensor(unittest.TestCase):
             setup_component(self.hass, 'sensor', config))
 
         assert len(self.hass.states.entity_ids()) == 0
+
+    @patch('os.stat', Mock(return_value=FAKE))
+    def test_valid_path(self):
+        """Test the filesize class."""
+        config = {
+            'sensor': {
+                'platform': 'filesize',
+                CONF_FILE_PATHS: ['/tmp']}
+        }
+
+        self.assertTrue(
+            setup_component(self.hass, 'sensor', config))
+
+        assert len(self.hass.states.entity_ids()) == 1
 
         #state = self.hass.states.get('sensor.test_api_streamspy')
 

